### DISCUSSION TITLE
Allow labels to be defined based on locale

### DISCUSSION THREAD

**Discussion by jmikrut** [payloadcms/payload#1262] [Roadmap]
A common request that we get is to allow field, collection, and global labels to be generated based on locale or other info. 

When we support [admin panel i18n](https://github.com/payloadcms/payload/discussions/300),  this will be huge, as it could allow Payload devs to generate labels for each admin panel locale—dramatically improving the UX of the admin UI for non-English speakers.

We will need to watch out for the ramifications that this has in GraphQL, as currently we are generating GraphQL schema names based on label, which also should be improved in parallel with this feature.

Marking as planned.

**Comment by quornik** [comment]
This is very interesting approach and I would love to see it being done. Right now, as I create PayloadCMS UI in polish, it effectively renders GraphQL  unuseable for me.

**Comment by Kikky** [comment]
Nice! Waiting for it! 👍

**Comment by reiv** [comment]
Not just the GraphQL schema; the generated TypeScript types also use the labels if available. It would be nice if we could set a `typeLocale` / `codegenLocale` or something like that, which for most people will probably be `en`.

**Comment by jmikrut** [comment]
This is now complete!
